src/mono/mono/eglib/gdate-unix.c

```diff
@@ -65,6 +65,12 @@ g_usleep (gulong microseconds)
 
 	do {
 		ret = clock_nanosleep (CLOCK_MONOTONIC, TIMER_ABSTIME, &target, NULL);
+#if HOST_ANDROID
+		// Workaround for incorrect implementation of clock_nanosleep return value on old Android (<=5.1)
+		// See https://github.com/xamarin/xamarin-android/issues/6600
+		if (ret == -1)
+			ret = errno;
+#endif
```

Comment:
@simonrozsival What if we added a declaration for `gint g_clock_nanosleep (clockid_t clockid, gint flags, const struct timespec *request, struct timespec *remain)` (under appropriate ifdefs) to `glib.h` and had less copy/pasted code.

Reply:
you're right, it'll be cleaner 👍